aws/resource_aws_codedeploy_deployment_group.go

```diff
@@ -630,6 +896,85 @@ func alarmConfigToMap(config *codedeploy.AlarmConfiguration) []map[string]interf
+func deploymentStyleToMap(style *codedeploy.DeploymentStyle) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	if style == nil {
+		return result
+	}
+
+	item := make(map[string]interface{})
+	item["deployment_option"] = *style.DeploymentOption
+	item["deployment_type"] = *style.DeploymentType
+	result = append(result, item)
+	return result
+}
+
+// loadBalancerInfoToMap converts a codedeploy.LoadBalancerInfo object
+// into a []map[string]interface{} list containing a single item
+func loadBalancerInfoToMap(loadBalancerInfo *codedeploy.LoadBalancerInfo) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, 1)
+
+	if loadBalancerInfo == nil {
+		return result
+	}
+
+	elbs := make([]interface{}, 0, len(loadBalancerInfo.ElbInfoList))
+	for _, elb := range loadBalancerInfo.ElbInfoList {
+		item := make(map[string]interface{})
+		item["name"] = *elb.Name
+		elbs = append(elbs, item)
+	}
+
+	targetGroups := make([]interface{}, 0, len(loadBalancerInfo.TargetGroupInfoList))
+	for _, targetGroup := range loadBalancerInfo.TargetGroupInfoList {
+		item := make(map[string]interface{})
+		item["name"] = *targetGroup.Name
+		targetGroups = append(targetGroups, item)
+	}
+
+	lbInfo := make(map[string]interface{})
+	lbInfo["elb_info"] = schema.NewSet(loadBalancerInfoHash, elbs)
+	lbInfo["target_group_info"] = schema.NewSet(loadBalancerInfoHash, targetGroups)
+	result = append(result, lbInfo)
+
+	return result
+}
+
+// blueGreenDeploymentConfigToMap converts a codedeploy.BlueGreenDeploymentConfiguration object
+// into a []map[string]interface{} list containing a single item
+func blueGreenDeploymentConfigToMap(config *codedeploy.BlueGreenDeploymentConfiguration) []map[string]interface{} {
+	list := make([]map[string]interface{}, 0)
+
+	if config == nil {
+		return list
+	}
+
+	m := make(map[string]interface{})
+
+	a := make([]map[string]interface{}, 0)
+	deploymentReadyOption := make(map[string]interface{})
+	deploymentReadyOption["action_on_timeout"] = *config.DeploymentReadyOption.ActionOnTimeout
+	deploymentReadyOption["wait_time_in_minutes"] = *config.DeploymentReadyOption.WaitTimeInMinutes
+	m["deployment_ready_option"] = append(a, deploymentReadyOption)
+
+	b := make([]map[string]interface{}, 0)
+	greenFleetProvisioningOption := make(map[string]interface{})
+	greenFleetProvisioningOption["action"] = *config.GreenFleetProvisioningOption.Action
+	m["green_fleet_provisioning_option"] = append(b, greenFleetProvisioningOption)
+
+	c := make([]map[string]interface{}, 0)
+	blueInstanceTerminationOption := make(map[string]interface{})
+	blueInstanceTerminationOption["action"] = *config.TerminateBlueInstancesOnDeploymentSuccess.Action
+	blueInstanceTerminationOption["termination_wait_time_in_minutes"] = *config.TerminateBlueInstancesOnDeploymentSuccess.TerminationWaitTimeInMinutes
+	m["terminate_blue_instances_on_deployment_success"] = append(c, blueInstanceTerminationOption)
+
```

Comment:
Are you sure that none of these options will ever be `nil`? Otherwise this would cause a crash here.